packages/tools/test/EraserTool_test.js

```diff
@@ -0,0 +1,208 @@
+        if (el.parentNode) {
+          el.parentNode.removeChild(el);
+        }
+      });
+    });
+
+    it('Should successfully delete a length annotation on a canvas with mouse down - 512 x 128', function (done) {
+      const element = createViewport(
+        this.renderingEngine,
+        ViewportType.STACK,
+        512,
+        128
+      );
+
+      this.DOMElements.push(element);
+
+      const imageId1 = 'fakeImageLoader:imageURI_64_64_10_5_1_1_0';
+      const vp = this.renderingEngine.getViewport(viewportId);
+
+      const addEventListenerForAnnotationRemoved = () => {
+        eventTarget.addEventListener(csToolsEvents.ANNOTATION_REMOVED, () => {
+          const lengthAnnotations = annotation.state.getAnnotations(
+            LengthTool.toolName,
+            element
+          );
+          expect(lengthAnnotations).toBeDefined();
+          expect(lengthAnnotations.length).toBe(0);
+          done();
+        });
+      };
+
+      element.addEventListener(Events.IMAGE_RENDERED, () => {
+        const index1 = [32, 32, 0];
+        const index2 = [10, 1, 0];
+
+        const { imageData } = vp.getImageData();
+
+        const {
+          pageX: pageX1,
+          pageY: pageY1,
+          clientX: clientX1,
+          clientY: clientY1,
+          worldCoord: worldCoord1,
+        } = createNormalizedMouseEvent(imageData, index1, element, vp);
+        const { worldCoord: worldCoord2 } = createNormalizedMouseEvent(
+          imageData,
+          index2,
+          element,
+          vp
+        );
+
+        const camera = vp.getCamera();
+        const { viewPlaneNormal, viewUp } = camera;
+
+        const lengthAnnotation = {
+          highlighted: true,
+          invalidated: true,
+          metadata: {
+            toolName: LengthTool.toolName,
+            viewPlaneNormal: [...viewPlaneNormal],
+            viewUp: [...viewUp],
+            FrameOfReferenceUID: vp.getFrameOfReferenceUID(),
+            referencedImageId: imageId1,
+          },
+          data: {
+            handles: {
+              points: [[...worldCoord1], [...worldCoord2]],
+              activeHandleIndex: null,
+              textBox: {
+                hasMoved: false,
+                worldPosition: [0, 0, 0],
+                worldBoundingBox: {
+                  topLeft: [0, 0, 0],
+                  topRight: [0, 0, 0],
+                  bottomLeft: [0, 0, 0],
+                  bottomRight: [0, 0, 0],
+                },
+              },
+            },
+            label: '',
+            cachedStats: {},
+          },
+        };
+        annotation.state.addAnnotation(lengthAnnotation, element);
+
+        let evt = new MouseEvent('mousedown', {
+          target: element,
+          buttons: 1,
+          clientX: clientX1,
+          clientY: clientY1,
+          pageX: pageX1,
+          pageY: pageY1,
+        });
+
+        addEventListenerForAnnotationRemoved();
+        document.dispatchEvent(evt);
```

Comment:
Dispatch the event on the viewport element itself. (i.e. `element.dispatchEvent(evt)`)